devices/Rtc/Devices/Ds1302/Ds1302.cs

```diff
@@ -0,0 +1,186 @@
+        public Ds1302(GpioController controller, int clockPin, int dataPin, int resetPin)
+        {
+            _controller = controller ?? throw new ArgumentNullException();
+            _clockPin = _controller?.OpenPin(clockPin, PinMode.Output);
+            _resetPin = _controller?.OpenPin(resetPin, PinMode.Output);
+            _dataPin = _controller?.OpenPin(dataPin, PinMode.Input);
+
+            _resetPin?.Write(PinValue.Low);
+            _clockPin?.Write(PinValue.Low);
+        }
+
+        /// <summary>
+        /// Checks if the clock is halted.
+        /// </summary>
+        /// <returns><b>true</b> when clock is halted, else <b>false</b></returns>
+        public bool IsHalted()
+        {
+            PrepareRead(Ds1302Registers.REG_SECONDS);
+            byte seconds = ReadByte();
+            EndTransmission();
+            return (byte)(seconds & 0b10000000) > 0;
+        }
+
+        /// <summary>
+        /// Halts the clock.
+        /// </summary>
+        public void Halt()
+        {
+            PrepareWrite(Ds1302Registers.REG_SECONDS);
+            WriteByte(0b10000000);
+            EndTransmission();
+        }
+
+        /// <summary>
+        /// Read time from the device.
+        /// </summary>
+        /// <returns>Time from the device.</returns>
+        protected override DateTime ReadTime()
+        {
+            PrepareRead(Ds1302Registers.REG_BURST);
+            var seconds = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b01111111));
+            var minutes = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b01111111));
+            var hours = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b00111111));
+            var days = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b00111111));
+            var months = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b00011111));
+            var dayOfWeek = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b00000111));
+            var years = NumberHelper.Bcd2Dec((byte)(ReadByte() & 0b01111111));
+
+            EndTransmission();
+
+            return new DateTime(2000 + years, months, days, hours, minutes, seconds, 0);
+        }
+
+        /// <summary>
+        /// Sets date and time.
+        /// </summary>
+        /// <param name="time">Date and time.</param>
+        protected override void SetTime(DateTime time)
+        {
+            PrepareWrite(Ds1302Registers.REG_WP);
+            WriteByte(0b00000000);
+            EndTransmission();
+
+            PrepareWrite(Ds1302Registers.REG_BURST);
+            WriteByte(NumberHelper.Dec2Bcd(time.Second));
+            WriteByte(NumberHelper.Dec2Bcd(time.Minute));
+            WriteByte(NumberHelper.Dec2Bcd(time.Hour));
+            WriteByte(NumberHelper.Dec2Bcd(time.Day));
+            WriteByte(NumberHelper.Dec2Bcd(time.Month));
+            WriteByte(NumberHelper.Dec2Bcd((int)time.DayOfWeek));
+            WriteByte(NumberHelper.Dec2Bcd(time.Year - 2000));
+
+            WriteByte(0b1000000);
```

Comment:
```suggestion
            WriteByte(0b0100_0000);
```

Same easier to read and please check that it's the proper one as I've adjusted to add a 0 in front to have 2 groups of 4. And as always, a quick comment for the magic number :-)